src/quartz/tasograph/tasograph.cpp

```diff
@@ -2371,9 +2389,9 @@ bool Graph::_pattern_matching(
       }
     }
   }
-  if (!fail) {
-    fail = !_loop_check_after_matching(xfer);
-  }
+  //   if (!fail) {
+  //     fail = !_loop_check_after_matching(xfer);
+  //   }
```

Comment:
#119 may need to be updated accordingly here.

Reply:
Yep, this could be one of the cause of #119, but during our meeting we found that even if we fix this, bug in #119 still exists.